lms/templates/student_account/register.underscore

```diff
@@ -29,11 +29,17 @@
             <%
             _.each( context.providers, function( provider) {
                 if ( provider.registerUrl ) { %>
-                    <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
-                        <span class="icon fa <%- provider.iconClass %>" aria-hidden="true"></span>
-                        <span aria-hidden="true"><%- provider.name %></span>
-                        <span class="sr"><%- _.sprintf( gettext("Create account using %(providerName)s."), {providerName: provider.name} ) %></span>
-                    </button>
+                    <div class="button-container" <% if ( provider.buttonColor ) { %>style="color: <%- provider.buttonColor %>; background-color: <%- provider.buttonColor %>"<% } %>>
+                        <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
+                            <div class="icon <% if ( provider.iconClass ) { %>fa <%- provider.iconClass %><% } %>" <% if ( provider.buttonColor ) { %>style="background: <%- provider.buttonColor %>"<% } %> aria-hidden="true">
+                                <% if ( provider.iconImage ) { %>
+                                    <img src="<%- provider.iconImage %>" alt="<%- provider.name %> icon" />
+                                <% } %>
+                            </div>
+                            <span <% if ( provider.textColor ) { %>style="color: <%- provider.textColor %>"<% } %> aria-hidden="true"><%- provider.name %></span>
```

Comment:
Adding the text colour as an inline value here does not allow for updating the hover state colour. I don't think there is any need to use inline styles for styling the button and its text It would be better to assign the button a custom class and style it via Sass/CSS so you can override the various button states while continuing to use stylesheets for all styling.